src/v2/guide/reactivity.md

```diff
@@ -4,80 +4,80 @@ type: guide
 
-## Async Update Queue
+## Hàng đợi cập nhật không đồng bộ
 
-In case you haven't noticed yet, Vue performs DOM updates **asynchronously**. Whenever a data change is observed, it will open a queue and buffer all the data changes that happen in the same event loop. If the same watcher is triggered multiple times, it will be pushed into the queue only once. This buffered de-duplication is important in avoiding unnecessary calculations and DOM manipulations. Then, in the next event loop "tick", Vue flushes the queue and performs the actual (already de-duped) work. Internally Vue tries native `Promise.then` and `MessageChannel` for the asynchronous queuing and falls back to `setTimeout(fn, 0)`.
+Trong trường hợp bạn chưa để ý, Vue thực hiện update DOM **bất đồng bộ**. Bất cứ khi nào thay đổi dữ liệu được quan sát thấy, nó sẽ mở một hàng đợi và đưa tất cả các thay đổi dữ liệu xảy ra trong cùng một vòng lặp sự kiện. Nếu cùng một `watcher` được kích hoạt nhiều lần, nó sẽ được đưa vào hàng đợi duy nhất một lần. Việc loại bỏ trùng lặp đệm này rất quan trọng trong việc tránh các phép tính không cần thiết và các thao tác DOM thừa thãi. Sau đó, trong vòng lặp sự kiện tiếp theo được "đánh dấu" (tick) và Vue sẽ xóa hàng đợi cũ và thực hiện công việc thực tế (đã loại đi các tác vụ trùng lặp). Để bạn hiểu rõ hơn thì Vue cố gắng thực hiện `Promise.then` và `MessageChannel` cho hàng đợi bất đồng bộ và trả về callback `setTimeout(fn, 0)`.
 
-For example, when you set `vm.someData = 'new value'`, the component will not re-render immediately. It will update in the next "tick", when the queue is flushed. Most of the time we don't need to care about this, but it can be tricky when you want to do something that depends on the post-update DOM state. Although Vue.js generally encourages developers to think in a "data-driven" fashion and avoid touching the DOM directly, sometimes it might be necessary to get your hands dirty. In order to wait until Vue.js has finished updating the DOM after a data change, you can use `Vue.nextTick(callback)` immediately after the data is changed. The callback will be called after the DOM has been updated. For example:
+Ví dụ, khi bạn thiết lập `vm.someData = 'new value'`, component sẽ không re-render ngay lập tức. Nó sẽ được update ở lần `tick` tiếp theo, khi hàng đợi bị xóa. Hầu hết thời gian chúng ta không cần phải quan tâm về điều này, nhưng nó có thể giúp ích khi bạn muốn làm điều gì đó phụ thuộc vào trạng thái của DOM sau khi cập nhật. Mặc dù Vue.js thường khuyến khích các nhà phát triển suy nghĩ theo cách "theo hướng dữ liệu" và tránh thao tác trực tiếp vào DOM, nhưng đôi khi chúng ta cũng nên thử điều gì đó mới mẻ. Để chờ cho đến khi Vue.js cập nhật xong DOM sau khi thay đổi dữ liệu, có thể sử dụng `Vue.nextTick(callback)` ngay lập tức sau khi thay đổi dữ liệu. Hàm callback sẽ được gọi sau khi DOM đã hoàn tất việc cập nhật. Ví dụ:
```

Comment:
* "re-render" là "render lại"
* "update" hoàn toàn có thể dịch là "cập nhật"
* "tick" không phải là code, không cần backtick
* "Most of the time" không dịch thành "hầu hết thời gian" mà đơn giản là "thường thì"
* "It can be tricky" không thể dịch thành "nó có thể giúp ích" được.
* Trong bài dịch này bạn thường sử dụng cụm từ "nhà phát triển." Mình thì nghĩ đơn giản là dùng "developer."
* "Sau khi DOM đã hoàn tất việc cập nhật" là không chính xác, vì DOM không thực hiện việc cập nhật. Đơn giản ở đây là "DOM đã được cập nhật xong."